recipes/olc-pge/all/conanfile.py

```diff
@@ -0,0 +1,72 @@
+from conans import ConanFile, tools, errors
+
+
+class oldPixelGameEngineConan(ConanFile):
+    name = "olc-pge"
+    homepage = "https://github.com/OneLoneCoder/olcPixelGameEngine"
+    description = "The olcPixelGameEngine is a single-file prototyping and game-engine framework created in C++. It is cross platform, compiling on Windows via Visual Studio and Code::Blocks, and on Linux with a modern g++."
+    topics = ("conan", "olc", "pge", "pixelengine", "pixelgameengine", "pgex",
+              "game-development", "game-engine", "engine", "gamedev", "gaming", "graphics")
+    license = "LicenseRef-LICENSE"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "image_loader": ['png', 'stb', 'default'],
+    }
+    default_options = {
+        "image_loader": 'default',
+    }
```

Comment:
I don't like auto or default options values, we don't clearly understand what is the backend, and it requires tricks in package_id.

I would do:

```python
    options = {
        "image_loader": ["png", "stb", "gdi"],
    }
    default_options = {
        "image_loader": "png",
    }

    def config_options(self):
        if self.settings.os == "Windows":
            self.options.image_loader = "gdi"

    def configure(self):
        if self.options.image_loader = "gdi" and self.settings.os != "Windows":
            raise ConanInvalidCondiguration("gdi image loader only supported on Windows")
```

Reply:
Yes, see the comment I've added in the code snippet